frontend/app/actions/ActionRefiners.scala

```diff
@@ -33,20 +42,73 @@ object Functions extends LazyLogging {
   def authenticated(onUnauthenticated: RequestHeader => Result = chooseRegister(_)): ActionBuilder[AuthRequest] =
     new AuthenticatedBuilder(AuthenticationService.authenticatedUserFor, onUnauthenticated)
 
-  def memberRefiner(onNonMember: RequestHeader => Result = notYetAMemberOn(_)) =
-    new ActionRefiner[AuthRequest, AnyMemberTierRequest] {
-      override def refine[A](request: AuthRequest[A]) = request.forMemberOpt {
-        _.map(member => MemberRequest(member, request)).toRight(onNonMember(request))
+  type SubRequestOrResult[A] = Future[Either[Result, SubReqWithSub[A]]]
+  type PaidSubRequestOrResult[A] = Future[Either[Result, SubscriptionRequest[A] with PaidSubscriber]]
```

Comment:
BTW, to make it easy to go from the `OptionT` that we get out of the for-comprehension to `EitherT`, we can use `toLeft` from `OptionT`: http://docs.typelevel.org/api/scalaz/nightly/index.html#scalaz.OptionT